File: AccountJava/app/src/main/java/com/example/accountjava/Account.java
package com.example.accountjava;

public class Account {
    public final String accountNumber;
    private int balance;

    public int getBalance(){
        return balance;
    }
    protected void setBalance(int newBalance){
        balance = newBalance;
    }

    public Account(String accountNumber, int balance) {
        this.accountNumber = accountNumber;
        this.balance = balance;
    }

    public Account(String accountNumber){
//        this.accountNumber = accountNumber;
//        this.balance = 0;     아래와 같다.

        this(accountNumber, 0);
    }
}

File: AccountKotlin/app/src/main/java/com/example/accountkotlin/Account.kt
package com.example.accountkotlin

import java.lang.NullPointerException

abstract class Account(accountNumber: String, balance: Int = 0) {
    val accountNumber  = accountNumber   //생성자에 있는 string 타입을 가져오므로 :string을 생략
    var balance = balance
        protected set(newBalance){
            println("${balance} -> ${newBalance}")
            field = newBalance
        }

    override fun toString() : String{
        return "${accountNumber} : ${balance} "
    }

    abstract fun aMonthHasPassed()
}